aten/src/ATen/native/utils/ParamsHash.h

```diff
@@ -38,5 +38,59 @@ struct ParamsEqual {
   }
 };
 
+// Provide explicit byte-for-byte constructors to avoid uwittingly leaving
+// padding bytes unitialized (e.g., when passing Params by value)
+template <typename T>
+struct ParamsWrapper {
+  T pod;
+  static_assert(std::is_standard_layout_v<T>, "ParamsWrapper cannot wrap non-POD data");
+
+  ParamsWrapper<T>() {
+    memset(&(this->pod), 0, sizeof(T));
+  }
+
+  ParamsWrapper<T>(const ParamsWrapper<T> &other) {
```

Comment:
Should this be `ParamsWrapper(const ParamsWrapper &other) {`? Similar for other ctors/functions.

https://cplusplus.com/forum/general/11522/

Reply:
Good catch, thanks!